tensorboard/loader.py

```diff
@@ -399,6 +401,254 @@ def __str__(self):
+
+    Raises:
+      ValueError: If offset is greater than size, or offset or size
+          decreased from the last invocation.
+
+    :type offset: int
+    :type size: int
+    """
+    if offset > size:
+      raise ValueError('offset (%d) can not exceed size (%d)' % (offset, size))
+    self._rate_offset.set_value(offset)
+    self._rate_size.set_value(size)
+    self._offset = offset
+    self._size = size
+    now = self._clock()
+    if not self._initialized:
+      self._last_log_time = now
+      self._last_bar_time = now
+      self._initialized = True
+      return
+    elapsed = now - self._last_log_time
+    if elapsed >= Progress.LOG_INTERVAL_SECONDS:
+      self._last_log_time = now
+      self._show_log()
+    elapsed = now - self._last_bar_time
+    if elapsed >= Progress.BAR_INTERVAL_SECONDS:
+      self._last_bar_time = now
+      self._show_bar()
+
+  def close(self):
+    """Forces progress to be written to log.
+
+    This method exists because we don't want the progress bar to say
+    something like 98% once the file is done loading.
+    """
+    self._show_log()
+    self._show_bar()
+    # Instructs util.LogHandler to clear the ephemeral logging state.
+    self._bar_callback('')
+
+  def sleep(self, seconds):
+    """Sleeps for a given number of seconds.
+
+    Time spent sleeping in this method does not have a detrimental
+    impact on the consumption rate.
+
+    :type seconds: float
+    """
+    self._sleep(seconds)
+    self._rate_offset.bump()
+
+  def _show_log(self):
+    self._log_callback('Loaded %s', self._get_message())
+
+  def _show_bar(self):
+    sofar = int(self._get_fraction() * Progress.BAR_WIDTH)
+    bar = (Progress.BLOCK_DARK * sofar +
+           Progress.BLOCK_LIGHT * (Progress.BAR_WIDTH - sofar))
+    self._bar_callback(u'%s %s ', bar, self._get_message())
+
+  def _get_message(self):
+    rate_offset = self._rate_offset.get_rate()  # summary processing speed
+    rate_size = self._rate_size.get_rate()  # summary production speed
+    message = u'%d%% of %s%s%s' % (
+      int(self._get_fraction() * 100.0),
+      _localize_int(self._size),
+      self._get_rate_suffix(Progress.DELTA, rate_offset),
+      self._get_rate_suffix(Progress.NABLA, rate_size))
+    if rate_offset and rate_size and rate_offset < rate_size:
+      # If TensorFlow is writing summaries to disk faster than we can
+      # insert them into the database, that's kind of problematic.
+      message += u' ' + self._make_red(u'[meltdown]')
+    return message
+
+  def _get_fraction(self):
+    if not self._size:
+      return 0.0
+    else:
+      return float(self._offset) / self._size
+
+  def _get_rate_suffix(self, symbol, rate):
+    if not rate:
+      return u''
+    return u' %s %sB/s' % (symbol, _localize_int(rate))
+
+  def _make_red(self, text):
+    return (util.Ansi.BOLD +
+            util.Ansi.RED +
+            (util.Ansi.FLIP if self._offset % 2 == 0 else u'') +
```

Comment:
When you say "in a really frightening way," do you mean, "whether it's black-red or red-black is essentially randomized each time that we set the progress bar, because the parity of each read is essentially random"? If so, I understand (but didn't expect that that was the behavior that you wanted); if not, I'm missing something.

Anyway, the meltdown text rendering behavior isn't super important to me; do whatever you want. :-)